packages/near-membrane-base/src/membrane.ts

```diff
@@ -1214,88 +1242,100 @@ export function init(
             return createPointer(proxy);
         },
         // callableApply
-        (
-            targetPointer: Pointer,
-            thisArgValueOrPointer: PrimitiveOrPointer,
-            ...listOfValuesOrPointers: PrimitiveOrPointer[]
-        ): PrimitiveOrPointer => {
-            targetPointer();
-            const fn = getSelectedTarget();
-            const thisArg = getLocalValue(thisArgValueOrPointer);
-            const args = listOfValuesOrPointers.map(getLocalValue);
-            let value;
-            try {
-                value = apply(fn, thisArg, args);
-            } catch (e) {
-                throw pushErrorAcrossBoundary(e);
-            }
-            return getTransferableValue(value);
-        },
+        instrumentCallableWrapper(
```

Comment:
these are the callable functions that are provided to the red side from blue, so they are considered inbound because red will invoke them when it needs to carry an operation on blue.